### DISCUSSION TITLE
Let swc_bundler run on the browser using wasm

### DISCUSSION THREAD

**Discussion by eng-cc** [swc-project/swc#7713] [Ideas]
I am working on a pure web code editor, hoping to compile and package Typescript code in the browser.

If we can abstract the part of reading and writing files in swc_bundler, we can use the external callback to get\set files, so that swc_bundler can be used in the browser.

I hope to get some advice on feasibility, if the solution is feasible, I plan to help doing it.

_similar_ question: https://github.com/swc-project/swc/discussions/7642

**Comment by kwonoj** [comment]
Please do not create duplicates with already answered.
